packages/atoms/src/Radio.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { NativeStyleObject, prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { TouchableOpacity, TouchableOpacityProps, View } from 'react-native';
+
+interface RadioProps extends Omit<TouchableOpacityProps, 'style' | 'onPress'> {
+    value: string;
+    isChecked?: boolean;
+    isDisabled?: boolean;
+    onPress?: (value: string) => void;
+    style?: NativeStyleObject;
+}
+
+type RadioStyleProps = {
+    isChecked: boolean;
+    isDisabled: boolean;
+};
+
+const radioStyle = prepareNativeStyle<RadioStyleProps>((utils, { isChecked, isDisabled }) => ({
+    height: 24,
+    width: 24,
+    backgroundColor: isDisabled ? utils.colors.gray400 : utils.colors.white,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderRadius: utils.borders.radii.round,
+    borderWidth: isChecked ? utils.borders.widths.lg : utils.borders.widths.md,
+    borderColor: utils.colors.gray400,
+    extend: {
+        condition: isChecked && !isDisabled,
+        style: { borderColor: utils.colors.green },
+    },
+}));
+
+const radioCheckStyle = prepareNativeStyle<Omit<RadioStyleProps, 'isChecked'>>(
+    (utils, { isDisabled }) => ({
+        height: 14,
+        width: 14,
+        borderRadius: utils.borders.radii.round,
+        backgroundColor: isDisabled ? utils.colors.gray600 : utils.colors.forest,
+    }),
+);
+
+export const Radio = ({
+    value,
+    isChecked = false,
+    onPress,
+    isDisabled = false,
+    style,
+    ...props
+}: RadioProps) => {
+    const { applyStyle } = useNativeStyles();
+
+    const triggerRadioChange = () => {
+        if (!isDisabled) {
+            onPress?.(value);
+        }
```

Comment:
Nitpick for discussion. I prefer this kind of guard

```suggestion
        if (isDisabled) return
        onPress?.(value);
```

Reply:
Also I am not a big fan of optional chaining just for the function call. I would like this to be included in the condition so once you pass the condition, you're just calling the function for sure.

Reply:
> 

In my opinion, both solutions (optional chaining for fnc and checking in condition) are on a similar level and depend on preferences and usage. Maybe this could be optimized like:

        <TouchableOpacity
            disabled={isDisabled}
            onPress={() => onPress?.(value)}

...and remove `triggerRadioChange` fnc completely because `TouchableOpacity` is enabled/disabled by `isDisabled` prop in current solution. If isDisabled is true, onPress is not emitted.

Reply:
Optional chaining is really nice language feature and we shoudn't avoid it, but my question is why is it even necessary? I think onPress prop should be required anyway because, Radio/TouchableOpacity without onPress doesn't make sense.

Also in case you want to keep this function, please rename it to handlePress/handleRadioPress

Reply:
> onPress prop should be required anyway because, Radio/TouchableOpacity without onPress doesn't make sense.

I agree. Fixed in https://github.com/trezor/trezor-suite/pull/5444/commits/d2e214e9ec467d9e772d97adb5db2d2e7ce8f398